python/tvm/hybrid/parser.py

```diff
@@ -72,15 +73,17 @@ def __init__(self, args, usage, func_name=None):
             The name of the function to be lowered; if not provided,
             the compiler will use the name in the AST
         """
-        self.args = args[:]
+        self.args = list(args)
         self.usage = usage.copy()
         self._args = {} # Dict maps arg name to actual arg instance (either a var or a buffer)
-        self.var_buffers = {} # Buffers formed by mutatble variables
         self.alloc_buffers = {} # Buffers formed by allocate instructions
         self.loops_above = {} # State variable that indicates loop levels above the current node
         self.var_consts = {} # Variables that are determined as readonly in previous stage
         self.func_name = func_name # The name of the function to be lowered
-        self.iter_axis = []
+        self.outputs = [] # Output tensors' name
+        self.side_effect = set() # Tensors with side effects
```

Comment:
Just a question, what's side_effect for?

Reply:
Later for in-place operation.